modules/backendscheduler/backendscheduler.go

```diff
@@ -0,0 +1,391 @@
+	j := s.work.GetJob(req.JobId)
+	if j == nil {
+		return nil, fmt.Errorf("job not found: %s", req.JobId)
+	}
+
+	switch req.Status {
+	case tempopb.JobStatus_JOB_STATUS_SUCCEEDED:
+		j.Complete()
+		metricJobsCompleted.WithLabelValues(j.JobDetail.Tenant, j.JobDetail.Tenant).Inc()
+		level.Info(log.Logger).Log("msg", "job completed", "job_id", req.JobId)
+
+		// TODO: update the blocklist?
+
+	case tempopb.JobStatus_JOB_STATUS_FAILED:
+		j.Fail()
+		metricJobsFailed.WithLabelValues(j.JobDetail.Tenant, j.JobDetail.Tenant).Inc()
+		level.Error(log.Logger).Log("msg", "job failed", "job_id", req.JobId, "error", req.Error)
+
+	default:
+		return nil, fmt.Errorf("invalid job status: %v", req.Status)
+	}
+
+	return &tempopb.UpdateJobStatusResponse{
+		Success: true,
+	}, nil
+}
+
+// ListJobs returns all jobs for a given tenant
+func (s *BackendScheduler) ListJobs(_ context.Context) []*work.Job {
+	return s.work.Jobs()
+}
+
+// CreateCompactionJob creates a new compaction job for the given tenant and blocks.
+// Must be called under jobsMtx lock.
+func (s *BackendScheduler) createCompactionJob(_ context.Context, tenantID string, input []string) error {
+	// Skip blocks which already have a job
+	for _, blockID := range input {
+		for _, j := range s.work.Jobs() {
+			if j.JobDetail.Tenant == tenantID {
+				switch j.Type {
+				case tempopb.JobType_JOB_TYPE_COMPACTION:
+					for _, b := range j.JobDetail.Compaction.Input {
+						if b == blockID {
+							// TODO: consider continue when we have a stripe of like-blocks.
+							return nil
+						}
+					}
+				default:
+					continue
+				}
+			}
+		}
+	}
+
+	jobID := uuid.New().String()
+
+	job := &work.Job{
+		ID:   jobID,
+		Type: tempopb.JobType_JOB_TYPE_COMPACTION,
+		JobDetail: tempopb.JobDetail{
+			Tenant: tenantID,
+			Compaction: &tempopb.CompactionDetail{
+				Input: input,
+			},
+		},
+	}
+
+	err := s.work.AddJob(job)
+	if err != nil {
+		return fmt.Errorf("failed to create job: %w", err)
+	}
+
+	// Update metrics
+	metricJobsCreated.WithLabelValues(tenantID, job.Type.String()).Inc()
+	metricJobsActive.WithLabelValues(tenantID, job.Type.String()).Inc()
+
+	return nil
+}
+
+func (s *BackendScheduler) Owns(_ string) bool {
+	return true
+}
+
+func (s *BackendScheduler) StatusHandler(w http.ResponseWriter, _ *http.Request) {
+	x := table.NewWriter()
+	x.AppendHeader(table.Row{"tenant", "blocks", "status", "worker", "created", "start", "end"})
+
+	for _, j := range s.work.Jobs() {
+		x.AppendRows([]table.Row{
+			{j.JobDetail.Tenant, j.JobDetail.Compaction.Input, j.Status(), j.WorkerID(), j.CreatedTime(), j.StartTime(), j.EndTime()},
+		})
+	}
+
+	x.AppendSeparator()
+
+	// x.SetOutputMirror(w)
+	w.WriteHeader(http.StatusOK)
+	// w.Header().Set("Content-Type", "plain/text")
+	_, _ = io.WriteString(w, x.Render())
+}
+
+func (s *BackendScheduler) compactions(ctx context.Context) []tempopb.JobDetail {
```

Comment:
reading this func and they way its used in above in ScheduleOnce we have to be careful. this is going to create A LOT of jobs and repeatedly throw them away. i'm also not convinced of the value of storing every pending compaction job in the job tracker struct. 

i think we either need to:

- generate the job on demand. so create a job when `Next` is called. this to me feels the cleanest
- find a way to store a smaller number of "todo" jobs.

I prefer the former b/c we need to persist the work in flight to object storage and we don't want to persist "pending" jobs. only the ones that are in flight.